ape/loader-elf.S

```diff
@@ -213,7 +213,18 @@ macho:	.long	0xFEEDFACE+1
 //
 //	@see	APE_LOADER_ENTRY
 //	@see	ape/loader.h
-_start:	mov	%rsp,%rsi
+_start:
+
+//	Hack for detecting M1 Rosetta environment.
+//	https://github.com/jart/cosmopolitan/issues/429#issuecomment-1166704377
+	cmp	$-1,%ebx
+	jne	0f
+	cmp	$+1,%edx
+	jne	0f
+	mov	$XNU,%dl
+	mov	$0,%rcx
```

Comment:
You could save 5 bytes of code size here by encoding the operation as follows:

```asm
   0:   48 c7 c1 00 00 00 00    mov    $0x0,%rcx
   7:   31 c9                   xor    %ecx,%ecx
```

With x86-64 assembly, operations on the 32-bit versions of registers (i.e. `%e*` vs. `%r*`) will always clear the top bits as a side-effect. The 32-bit versions of instructions are usually shorter too, because they don't need to use a REX byte (i.e. 0x4x).